wolox-books/src/app/services/user.service.ts

```diff
@@ -21,4 +21,10 @@ export class UserService {
   loginUser(userData) {
     return this.http.post(this.API_URLS.login, userData, { observe: 'response' });
   }
+
+  isLogged() {
+    const headers = ["accessToken", "client", "uid"];
+    const clientHeaders = Object.keys(localStorage).sort();
+    return JSON.stringify(clientHeaders) === JSON.stringify(headers) && Object.values(localStorage).every(val => !!val);
```

Comment:
I didn't want to do just that becuase in that way I would just check if there's a key in localStorage called "accessToken" no matter which value has (in fact, it could be blank and I wouldn't had realized because I'm just checking for a key), and this particular token-auth mecanism of this rails backend has 3 headers that have to be all together in order to make the auth work (that's how it prevents this situation). So I'm checking with all the validation, that not just 'accessToken', but also 'client' and 'uid' headers are present, and its value is different than undefined/null (that's what the `val => !!val` does)

Reply:
You could set accessToken or other key just when the auth work, and clean it when auth fail, also you can keep other keys if you need them, but in this way you will avoid de long condition.

Reply:
I made it more "strict" just to ensure that someone doesn't access everything just by putting manually a localStorage key in the browser and then see everything like it was logged in. But of course this depends on the project/moment needs, this is more secure but also more long as you say

Reply:
There are a couple of ways we solve that in a project. The most common and useful is to handle the `401` error code. Because as you say, you can store some kind of `accessToken` to the local storage and you are good to go, but that key will be invalid so there's no content to show since the app will redirect you to the login/register screens 😉